service/src/main/java/tds/session/web/endpoints/HealthController.java

```diff
@@ -0,0 +1,18 @@
+package tds.session.web.endpoints;
+
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/session")
+public class HealthController {
+    @RequestMapping(value = "/isAlive", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    ResponseEntity<String> isAlive() {
```

Comment:
No, Spring's magic can handle non private classes/methods (I believe it uses reflection).  The same can be done for any Spring class.  For example, take a look at SessionServiceImpl.  The implementation is wired into the controller however the implementation is package private. 

Two controllers can have the same class level @RequestMapping but methods cannot.  For example, if SessionController had a `@RequestMapping(value = "/isAlive", method = RequestMethod.GET)` Spring would complain.